kong/runloop/certificate.lua

```diff
@@ -371,11 +374,159 @@ local function get_ca_certificate_store(ca_ids)
 end
 
 
+-- ordinary entities that reference ca certificates
+-- the value denotes which cache (kong.cache or kong.core_cache) is used
+local CA_CERT_REFERENCE_ENTITIES = {
+  "services",
+}
+
+-- plugins that reference ca certificates
+-- Format:
+-- mtls-auth = true
+local CA_CERT_REFERENCE_PLUGINS = {
+}
+
+local loaded_plugins
+local reference_plugins
+
+-- Examples:
+-- gen_iterator("services")
+-- gen_iterator("plugins", { mtls-auth = true })
+-- gen_iterator("plugins", "mtls-auth")
+-- We assume the field name is always `ca_certificates`
+local function gen_iterator(entity, plugins)
+  local options = {
+    workspace = null,
+  }
+
+  local iter = kong.db[entity]:each(1000, options)
```

Comment:
The `options` here can be configured separately, and if we are traversing plugins, we can use the "search_fields" feature to filter the plugins we need directly from the database level to improve the traversal performance. e.g. 

```
options = {
  workspace = null,
  search_fields = {
    name = plugin_name,
  }
}
```

Reply:
Good suggestion! done.

Reply:
Remove `search_fields` for now as this is EE-only. Will add it back in the EE PR.